<launch>
    
  <rosparam file="$(find roberto_hw_interface)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find roberto_hw_interface)/config/joint_limits.yaml" command="load"/>

  <node name="RobertoHardwareInterface" pkg="roberto_hw_interface" type="roberto_hw" output="screen"/>

  <!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/> -->
    
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
            /Roberto/joints_update
            /Roberto/JointR_EffortController
            /Roberto/JointL_EffortController
        "/> 
<!--
  <node name="topic_convert" pkg="roberto_hw_interface" type="topic_convert">
    <remap from="output" to="Roberto/JointD_VelocityController/command"/>
  </node>
-->

  <node name="diff_drive" pkg="roberto_hw_interface" type="diff_drive" output="screen">
    <remap from="outputLeft" to="Roberto/JointR_EffortController/command"/>
    <remap from="outputRight" to="Roberto/JointL_EffortController/command"/>
    <param name="totalMod"    value="1.0"/>
    <param name="leftMod"  value="1.0"/>
    <param name="rightMod"  value="-2.25"/>
    <param name="turnMod"  value="1.0"/>
  </node>

</launch>
